lua/chatcommands/coinflip.lua

```diff
@@ -0,0 +1,89 @@
+			end
+		else
+			ply:ChatPrint("[red]Player "..string.Implode(" ",t).." not found")
+		end
+	end
+end,{global=true,throttle=true})
+
+function initCointFlip(ply,target,amount)
+	if ply:PS_HasPoints(amount) and target:PS_HasPoints(amount) and CoinFlips[ply:SteamID()] == nil then
+		-- Both players have enough points AND an existing coinflip doesn't exist.
+		ply:ChatPrint("[orange]"..to:Nick().." is receiving your coinflip request.")
+		CoinFlips[ply:SteamID()] = {target:SteamID(), amount}
+		timer.Simple(120, CoinFlipRequestTimeout, ply, target) -- 2 minutes for the timeout.
+	elseif CoinFlips[ply:SteamID()] ~= nil then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh already hae a filp goin!")
+	elseif not ply:PS_HasPoints(amount) then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh don't hae dooze points!")
+	elseif not target:PS_HasPoints(amount) then
+		-- Should a player be told another player doesn't have enough points?
+		ply:ChatPrint("[red]y'r bruv doesn't hae dooze points to coinflip with yeh!")
+	end
+end
+
+local function CoinFlipRequestTimeout(ply, target)
+	table.remove(CoinFlips, ply:SteamID())
+	ply:ChatPrint("[fbc]The other bruv don't want to play. Try again lates.")
+	target:ChatPrint("[fbc]Missed out on a coinflip from " .. ply:Nick() .. ".")
+end
+
+function checkCoinFlipRequest(target)
+	-- There is probably a better way to do this. I have always found this kind of coding nasty once you get a sip of Java 8.
+	coinflipFound = false
+	for i,j in pairs(CoinFlips) do
+		if j[1] == target:SteamID() then
+			-- Coinflip Request Found
+			coinflipFound = true
+			finishCoinFlip(i, target)
+		end
+	end
+	if not coinflipFound then
+		ply:ChatPrint("[red]What the hell r yeh doing bruv? yeh don't hae a coinflip request!")
+	end
+end
+
+function finishCoinFlip(fromID, to)
+	-- Self explanatory, 'from' is the initiator player who's ID is the key in the table, 'to' is the other player.
+	local from = nil
+	local amount = CoinFlips[fromID][2]
+	-- Get Player from ID
+	for _,ply in ipairs( player.GetAll() ) do
+		if(ply:SteamID() == fromID) then from = ply end
+	end
+	if(from == nil) then
+		to:ChatPrint("[red]The other bruv left, gotta cancel this coinflip!")
+	elseif from:PS_HasPoints(amount) and to:PS_HasPoints(amount) then
+		-- Final Check, make sure they have funds still
+		local heads = math.random(0, 1) -- the "request from" player is always Heads.
+		BotSayGlobal("[edgy]" .. from:Nick() .. "[fbc] flipped a coin worth [rainbow]" .. (amount * 2) .. "[fbc] against [gold]".. to:Nick().. "[fbc] and...... [rainbow]" .. ((heads == 1) and "Won" or "Lost") .."!")
+		from:ChatPrint("[fbc]You " .. ((heads == 1) and "Won" or "Lost") .. " " .. amount .. ".")
+		to:ChatPrint("[fbc]You " .. ((heads == 0) and "Won" or "Lost") .. " " .. amount .. ".")
+		-- Instead of taking the amount away from both and then giving the winner the amount x 2, simply remove/add here
+		if heads == 1 then
+			from:PS_GivePoints(amount)
```

Comment:
Take before Give, in case Take errors. If you Give before Take errors, you've got a duping exploit